insights/client/data_collector.py

```diff
@@ -257,6 +250,65 @@ def run_collection(self, conf, rm_conf, branch_info, blacklist_report):
         self._write_blacklist_report(blacklist_report)
         logger.debug('Metadata collection finished.')
 
+    def data_redaction(self, rm_conf):
+        '''
+        Perform data redaction (password sed command and patterns),
+        write data to the archive in place
+        '''
+        logger.debug('Running content redaction...')
+
+        if not re.match(r'/var/tmp/.+/insights-.+', self.archive.archive_dir):
+            # sanity check to make sure we're only modifying
+            #   our own stuff in temp
+            # we should never get here but just in case
+            raise RuntimeError('ERROR: invalid Insights archive temp path')
+
+        if rm_conf is None:
+            rm_conf = {}
+        exclude = None
+        regex = False
+        if rm_conf:
+            try:
+                exclude = rm_conf['patterns']
+                if isinstance(exclude, dict) and exclude['regex']:
+                    # if "patterns" contains a non-empty "regex" dict
+                    logger.debug('Using regular expression matching for patterns.')
+                    exclude = exclude['regex']
+                    regex = True
+                logger.warn("WARNING: Skipping patterns defined in blacklist configuration")
+            except LookupError:
+                # either "patterns" was undefined in rm conf, or
+                #   "regex" was undefined in "patterns"
+                exclude = None
+        if not exclude:
+            logger.debug('Patterns section of blacklist configuration is empty.')
+
+        for dirpath, dirnames, filenames in os.walk(self.archive.archive_dir):
+
+            for f in filenames:
+                fullpath = os.path.join(dirpath, f)
+
+                logger.debug('Processing %s...', fullpath)
+
+                # password removal
+                sedcmd = Popen(['sed', '-rf', constants.default_sed_file, fullpath], stdout=PIPE)
```

Comment:
moving this logic to its own function would make it easier to test in isolation.